src/Conversion/InterventionImageFileConverter.php

```diff
@@ -0,0 +1,199 @@
+ */
+class InterventionImageFileConverter implements FileConverter
+{
+    public function supportsConversion(DBFile|string $from, string $toFormat, array $options = []): bool
+    {
+        $unsupportedOptions = $this->validateOptions($options);
+        if (!empty($unsupportedOptions)) {
+            return false;
+        }
+        // If backend is not InterventionBackend, return false
+        if ($from instanceof DBFile) {
+            $backend = $from->getImageBackend();
+            $from = $from->getExtension() ?: $from->getMimeType();
+        } else {
+            $backend = Injector::inst()->get(Image_Backend::class);
+        }
+        // This converter requires intervention image as the image backend
+        if (!is_a($backend, InterventionBackend::class)) {
+            return false;
+        }
+        /** @var InterventionBackend $backend */
+        $driver = $backend->getImageManager()->config['driver'] ?? null;
+        return $this->supportedByIntervention($from, $toFormat, $driver);
+    }
+
+    public function convert(DBFile $from, string $toFormat, array $options = []): DBFile
+    {
+        $problems = $this->validateOptions($options);
+        if (!empty($problems)) {
+            throw new FileConversionException('Invalid options provided: ' . implode(', ', $problems));
+        }
+        $quality = $options['quality'] ?? null;
+        try {
+            $result = $from->manipulateExtension(
+                $toFormat,
+                function (AssetStore $store, string $filename, string $hash, string $variant) use ($from, $quality) {
+                    $backend = clone $from->getImageBackend();
+                    if ($quality !== null) {
+                        $backend->setQuality($quality);
+                    }
+                    $config = ['conflict' => AssetStore::CONFLICT_USE_EXISTING];
+                    $tuple = $backend->writeToStore($store, $filename, $hash, $variant, $config);
+                    return [$tuple, $backend];
+                }
+            );
+        } catch (ImageException $e) {
+            throw new FileConversionException('Failed to convert: ' . $e->getMessage(), $e->getCode(), $e);
+        }
+        if ($result === null) {
+            throw new FileConversionException('File conversion resulted in null. Check whether original file actually exists.');
+        }
+        return $result;
+    }
+
+    private function validateOptions(array $options): array
+    {
+        $problems = [];
+        foreach ($options as $key => $value) {
+            if ($key !== 'quality') {
+                $problems[] = "unexpected option '$key'";
+                continue;
+            }
+            if (!is_int($value)) {
+                $problems[] = "quality value must be an integer";
+            }
+        }
+        return $problems;
+    }
+
+    private function supportedByIntervention(string $from, string $to, string $driver): bool
```

Comment:
Intervention Image doesn't currently tell you what it supports. [It might in the future](https://github.com/Intervention/image/issues/1326) but for now we have to check for ourselves.